pjsip/src/pjsua-lib/pjsua_media.c

```diff
@@ -4220,10 +4220,20 @@ static pj_status_t apply_med_update(pjsua_call_media *call_med,
 #endif
             }
             if (status != PJ_SUCCESS) {
+                if (call_med->type == PJMEDIA_TYPE_AUDIO) {
                 PJ_PERROR(1,(THIS_FILE, status,
                              "pjsua_aud_channel_update() failed "
-                                 "for call_id %d media %d",
+                             "for call_id %d media %dd",
                              call_id, mi));
+                }
+#if defined(PJMEDIA_HAS_VIDEO) && (PJMEDIA_HAS_VIDEO != 0)
+                else if (call_med->type == PJMEDIA_TYPE_VIDEO) {
+                PJ_PERROR(1,(THIS_FILE, status,
+                             "pjsua_vid_channel_update() failed "
+                             "for call_id %d media %d",
+                             call_id, mi));
+                }
+#endif
```

Comment:
For simplicity, alternative fix can be something like:
```
                 PJ_PERROR(1,(THIS_FILE, status,
                             "pjsua_%s_channel_update() failed "
                             "for call_id %d media %d",
                             (call_med->type==PJMEDIA_TYPE_AUDIO? "aud" : "vid")
                             call_id, mi));
```
Or
```
                 PJ_PERROR(1,(THIS_FILE, status,
                             "Failed updating %s "
                             "for call_id %d media %d",
                             (call_med->type==PJMEDIA_TYPE_AUDIO? "audio" : "video")
                             call_id, mi));
```

Reply:
yeah, this is much clearly, if there is no ``#if defined(PJMEDIA_HAS_VIDEO) && (PJMEDIA_HAS_VIDEO != 0)`` before. Besides I found ``enum pjmedia_type`` has more type defines other than audio and video.So, I though ``(call_med->type==PJMEDIA_TYPE_AUDIO? "audio" : "video")`` maybe lack some extension in future. How do you think?

Reply:
Good point.

With the upcoming text stream (currently in PR), I would prefer to get the media name, such as the one here:
https://github.com/pjsip/pjproject/blob/cd96f03d1e1338ca11a0c6f699ac1839fe6acfcf/pjsip/src/pjsua-lib/pjsua_media.c#L4270-L4272
and
https://github.com/pjsip/pjproject/blob/cd96f03d1e1338ca11a0c6f699ac1839fe6acfcf/pjsip/src/pjsua-lib/pjsua_media.c#L4115-L4116

Also, could you please help us change this one as well:
https://github.com/pjsip/pjproject/blob/cd96f03d1e1338ca11a0c6f699ac1839fe6acfcf/pjsip/src/pjsua-lib/pjsua_media.c#L4048-L4050

Reply:
Great, this is better solution.